<a href="https://colab.research.google.com/github/EkatAVD/EkatAVD/blob/main/AvdeevaEV.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
df = pd.DataFrame()

In [22]:
df=pd.read_csv("WorldCases2023.csv")

In [23]:
from pandas.core import describe
def df_info(data):
  print('-----------------------------------------------------------------')
  print(df.info())
  print('-----------------------------------------------------------------')
  print(df.isnull())
  print('-----------------------------------------------------------------')
  print(df.sample())
  print('-----------------------------------------------------------------')
  display(pd.DataFrame(round((df.isna().mean()*100),2)).style.background_gradient('coolwarm'))

Случаи заболевания Covid по странам - 2023 г.
Число зарегистрированных случаев и смертей от вируса SARS-CoV-2, вызывающего Covid-19, для каждой страны мира

Включает страну, общее количество случаев, новые случаи, общее количество смертей, новых детей, общее количество выздоровевших, новых выздоровевших, активных случаев, серьезных или критических, общее количество случаев на 1 миллион населения, смертей на 1 миллион населения, общее количество тестов, тестов на 1 миллион населения и население страны.

Данные по состоянию на 6 августа 2023 года.

In [6]:
df_info (df)

-----------------------------------------------------------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 232 entries, 0 to 231
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Rank               232 non-null    int64  
 1   Country            232 non-null    object 
 2   TotalCases         232 non-null    int64  
 3   NewCases           14 non-null     float64
 4   TotalDeaths        227 non-null    float64
 5   NewDeaths          3 non-null      float64
 6   TotalRecovered     210 non-null    float64
 7   NewRecovered       15 non-null     float64
 8   ActiveCases        212 non-null    float64
 9   SeriousCritical    123 non-null    float64
 10  Tot CasesPer1MPop  230 non-null    float64
 11  DeathsPer1MPop     225 non-null    float64
 12  TotalTests         213 non-null    float64
 13  TestsPer1MPop      213 non-null    float64
 14  Population         229 non-null    object 
dtypes: float

In [46]:
df.corr()

<ipython-input-46-2f6f6606aa2c>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()


Rank  TotalCases  NewCases  TotalDeaths  NewDeaths  \
Rank               1.000000   -0.457900 -0.474916    -0.408476       -1.0   
TotalCases        -0.457900    1.000000  0.987390     0.880589        1.0   
NewCases          -0.474916    0.987390  1.000000     0.983520        NaN   
TotalDeaths       -0.408476    0.880589  0.983520     1.000000        1.0   
NewDeaths         -1.000000    1.000000       NaN     1.000000        1.0   
TotalRecovered    -0.442693    0.999902  0.987826     0.891613        1.0   
NewRecovered      -0.376800    0.233860  0.999038    -0.039522        NaN   
ActiveCases       -0.304878    0.478603  0.473843     0.515295        1.0   
SeriousCritical   -0.348166    0.624985  0.984910     0.575560        NaN   
Tot CasesPer1MPop -0.173988    0.166081  0.097588     0.028631        1.0   
DeathsPer1MPop    -0.420366    0.218201  0.278838     0.286471        1.0   
TotalTests        -0.385602    0.838679  0.983663     0.798599        1.0   
TestsPer1MPop     -0.075747    0.053514  0.009802    -0.000501        1.0   

                   TotalRecovered  NewRecovered  ActiveCases  SeriousCritical  \
Rank                    -0.442693     -0.376800    -0.304878        -0.348166   
TotalCases               0.999902      0.233860     0.478603         0.624985   
NewCases                 0.987826      0.999038     0.473843         0.984910   
TotalDeaths              0.891613     -0.039522     0.515295         0.575560   
NewDeaths                1.000000           NaN     1.000000              NaN   
TotalRecovered           1.000000      0.233942     0.466692         0.639531   
NewRecovered             0.233942      1.000000     0.571973         0.156538   
ActiveCases              0.466692      0.571973     1.000000         0.181137   
SeriousCritical          0.639531      0.156538     0.181137         1.000000   
Tot CasesPer1MPop        0.187588      0.496028     0.051338         0.054871   
DeathsPer1MPop           0.236392     -0.254360     0.136604         0.163404   
TotalTests               0.849300     -0.084259     0.275961         0.553918   
TestsPer1MPop            0.070843     -0.218971    -0.048905        -0.020152   

                   Tot CasesPer1MPop  DeathsPer1MPop  TotalTests  \
Rank                       -0.173988       -0.420366   -0.385602   
TotalCases                  0.166081        0.218201    0.838679   
NewCases                    0.097588        0.278838    0.983663   
TotalDeaths                 0.028631        0.286471    0.798599   
NewDeaths                   1.000000        1.000000    1.000000   
TotalRecovered              0.187588        0.236392    0.849300   
NewRecovered                0.496028       -0.254360   -0.084259   
ActiveCases                 0.051338        0.136604    0.275961   
SeriousCritical             0.054871        0.163404    0.553918   
Tot CasesPer1MPop           1.000000        0.470486    0.092805   
DeathsPer1MPop              0.470486        1.000000    0.168928   
TotalTests                  0.092805        0.168928    1.000000   
TestsPer1MPop               0.548418        0.237262    0.203373   

                   TestsPer1MPop   
Rank                    -0.075747  
TotalCases               0.053514  
NewCases                 0.009802  
TotalDeaths             -0.000501  
NewDeaths                1.000000  
TotalRecovered           0.070843  
NewRecovered            -0.218971  
ActiveCases             -0.048905  
SeriousCritical         -0.020152  
Tot CasesPer1MPop        0.548418  
DeathsPer1MPop           0.237262  
TotalTests               0.203373  
TestsPer1MPop            1.000000

Вывели матрицу корреляции. Новые случаи заболевания  (NewCases/TotalCases)
 0.983520 имеют высокую положительную корреляцию в сравнении с общим числом заболевания.

In [24]:
df['NewCases'].sum()

3310.0

In [25]:
3310-1655

1655

В датасете есть строчка первая МИР поэтому сумм не логично считается, удалю эту строчку (О)

In [26]:
df.head()

Rank  Country  TotalCases  NewCases  TotalDeaths  NewDeaths  \
0     0    World   692616455    1655.0    6904586.0        6.0   
1     1      USA   107548847    1370.0    1170023.0        4.0   
2     2    India    44995933       NaN     531918.0        NaN   
3     3   France    40138560       NaN     167642.0        NaN   
4     4  Germany    38428685       NaN     174352.0        NaN   

   TotalRecovered  NewRecovered  ActiveCases  SeriousCritical  \
0     664825894.0       51336.0   20885975.0          37175.0   
1     105679821.0        2021.0     699003.0            710.0   
2      44462462.0           NaN       1553.0              NaN   
3      39970918.0           NaN          0.0            869.0   
4      38240600.0           NaN      13733.0              NaN   

   Tot CasesPer1MPop  DeathsPer1MPop    TotalTests  TestsPer1MPop   \
0            88856.0           885.8           NaN             NaN   
1           321228.0          3495.0  1.181184e+09       3527974.0   
2            31988.0           378.0  9.307980e+08        661721.0   
3           612013.0          2556.0  2.714902e+08       4139547.0   
4           458119.0          2078.0  1.223324e+08       1458359.0   

       Population  
0             NaN  
1    334,805,269   
2  1,406,631,776   
3     65,584,518   
4     83,883,596

In [30]:
df.drop(0, axis=0, inplace=True)

In [31]:
df.head()

Rank  Country  TotalCases  NewCases  TotalDeaths  NewDeaths  \
1     1      USA   107548847    1370.0    1170023.0        4.0   
2     2    India    44995933       NaN     531918.0        NaN   
3     3   France    40138560       NaN     167642.0        NaN   
4     4  Germany    38428685       NaN     174352.0        NaN   
5     5   Brazil    37728415       NaN     704794.0        NaN   

   TotalRecovered  NewRecovered  ActiveCases  SeriousCritical  \
1     105679821.0        2021.0     699003.0            710.0   
2      44462462.0           NaN       1553.0              NaN   
3      39970918.0           NaN          0.0            869.0   
4      38240600.0           NaN      13733.0              NaN   
5      36249161.0           NaN     774460.0              NaN   

   Tot CasesPer1MPop  DeathsPer1MPop    TotalTests  TestsPer1MPop   \
1           321228.0          3495.0  1.181184e+09       3527974.0   
2            31988.0           378.0  9.307980e+08        661721.0   
3           612013.0          2556.0  2.714902e+08       4139547.0   
4           458119.0          2078.0  1.223324e+08       1458359.0   
5           175193.0          3273.0  6.377617e+07        296146.0   

       Population  
1    334,805,269   
2  1,406,631,776   
3     65,584,518   
4     83,883,596   
5    215,353,593

In [32]:
df['NewCases'].sum()

1655.0

Теперь можно сказать сколько случаев смертей Covid-19 в мире

In [45]:
df['TotalDeaths'].sum()

6904586.0

In [42]:
df['TotalDeaths'].mean()


30551.265486725664

Вычислили ср. значение смертей по странам от Covid-19

In [61]:
df['TotalRecovered'].median()

228805.0

Вычислили  медиану по столбцу (всего востановлено)

In [62]:
df['SeriousCritical'].std()

131.82951721650963

Вычислили стандартное отклонение по столбцу (серьезно, критический) от COVID-19

In [47]:
df['Country'].unique()

array(['USA', 'India', 'France', 'Germany', 'Brazil', 'Japan', 'S. Korea',
       'Italy', 'UK', 'Russia', 'Turkey', 'Spain', 'Australia', 'Vietnam',
       'Taiwan', 'Argentina', 'Netherlands', 'Mexico', 'Iran',
       'Indonesia', 'Poland', 'Colombia', 'Greece', 'Austria', 'Portugal',
       'Ukraine', 'Chile', 'Malaysia', 'Israel', 'Belgium', 'DPRK',
       'Thailand', 'Canada', 'Czechia', 'Peru', 'Switzerland',
       'Philippines', 'South Africa', 'Romania', 'Denmark', 'Hong Kong',
       'Sweden', 'Serbia', 'Singapore', 'Iraq', 'New Zealand', 'Hungary',
       'Bangladesh', 'Slovakia', 'Georgia', 'Jordan', 'Ireland',
       'Pakistan', 'Norway', 'Finland', 'Kazakhstan', 'Slovenia',
       'Lithuania', 'Bulgaria', 'Morocco', 'Croatia', 'Guatemala',
       'Costa Rica', 'Lebanon', 'Bolivia', 'Tunisia', 'Cuba', 'Ecuador',
       'UAE', 'Panama', 'Uruguay', 'Mongolia', 'Nepal', 'Belarus',
       'Latvia', 'Saudi Arabia', 'Azerbaijan', 'Paraguay', 'Bahrain',
       'Sri Lanka', 'Domin

In [59]:
df['Country'].loc[15]

'Taiwan'

In [68]:
print(df['TotalCases']>df['TotalDeaths'])

1       True
2       True
3       True
4       True
5       True
       ...  
227     True
228    False
229    False
230     True
231     True
Length: 231, dtype: bool


In [78]:
print(df['NewDeaths']>1)

1       True
2      False
3      False
4      False
5      False
       ...  
227    False
228    False
229    False
230    False
231    False
Name: NewDeaths, Length: 231, dtype: bool
